packages/legend-graph/src/models/metamodels/pure/packageableElements/mapping/PropertyMapping.ts

```diff
@@ -70,35 +73,37 @@ export abstract class PropertyMapping implements Hashable {
    *
    * TODO: change this to use `SetImplemenetationReference`
    */
-  sourceSetImplementation: SetImplementation;
+  sourceSetImplementation: SetImplementationReference;
   /**
    * NOTE: in Pure, we actually only store `targetId` and `sourceId` instead of the
    * reference but for convenience and graph completeness validation purpose we will
    * resolve to the actual set implementations here
    *
    * TODO: change this to use `OptionalSetImplemenetationReference`
    */
-  targetSetImplementation?: SetImplementation | undefined;
+  targetSetImplementation?: OptionalSetImplementationReference;
```

Comment:
Since you have already used `OptionalSetImplementationReference`, there's no need to do `?:`

```suggestion
  targetSetImplementation: OptionalSetImplementationReference;
```

Reply:
And also, let's get rid of the comments 🙏